torch/csrc/autograd/engine.cpp

```diff
@@ -724,7 +724,7 @@ void validate_outputs(
 
     if (!metadata.is_same_shape(grad)) {
       if (metadata.is_expandable_to_shape(grad)) {
-        grad = at::sum_to(std::move(grad), metadata.shape());
+        grad = metadata.expand_grad(grad);
```

Comment:
This should actually be named `reduce grad` since `sum_to` performs a reduce.

Also whoops, this should've been in our earlier refactor PR, oh well, no action needed!